### ISSUE TITLE
When drawing shapes in/on a Grid the clipping at edges and corners is different from left/Top and Right/Bottom.

### ISSUE THREAD

**Issue by groovykool** [microsoft/microsoft-ui-xaml#2932]
**Describe the bug**
When drawing shapes in/on a Grid the clipping at edges and corners is different from left/Top and Right/Bottom.

**Steps to reproduce the bug**
Draw a shape at 0,0 and Gridwidth, GridHeight.
Repo: https://github.com/groovykool/DottedGrids.git

**Expected behavior**
Clipping in each corner and side of the grid should be the same.  And a line drawn on the edge/corner of a grid should be visible

**Screenshots**
![Screencap](https://user-images.githubusercontent.com/48837150/87722940-994d2d00-c77e-11ea-884c-0dce3123c2fa.png)

**Version Info**
2.4.2

NuGet package version: 
2.4.2

1909

Desktop

**Comment by chingucoding** [comment]
That's a really strange bug. Setting margin/padding etc doesn't change the behavior. Seems that grid really doesn't like rendering items that extend to the right or bottom.

**Comment by StephenLPeters** [comment]
very strange.. @codendone do you think this is a rendering issue?

**Comment by codendone** [comment]
I suspect there is an oddity in a rendering optimization here, with the code checking somewhere to not draw beyond the available bounds provided by its panel but not doing a similar check for negative values. The code should be changed to be consistent.

**Comment by groovykool** [comment]
Should I give up on Drawing on Grids and create a custom panel?

This is still an issue in 2.5

**Comment by groovykool** [comment]
Since WinUI 3 is not going to be available for UWP.  This bug needs to be fixed in WinUI 2.x.

**Comment by groovykool** [comment]
Will this Bug ever be fixed?  Still in 2.6.1.

**Comment by codendone** [comment]
Correct, this is an issue in core XAML code for determining whether a layout clip needs to be applied, and can only be fixed in a core system XAML change.

Apps can work around the issue by applying an explicit `UIElement.Clip` on the panel, though this requires assigning the correct size on the `RectangleGeometry` specified for the Clip.

**Comment by groovykool** [comment]
How can I report this bug for "core system XAML code".  Or Please forward this bug to the appropriate Microsoft Team.

I am very disappointed that it is over a year later and the bug is not even assigned to the correct team.

**Comment by StephenLPeters** [comment]
This is the correct team, however our bar for issues we fix with the OS version of the framework is quite high since we are investing in Winui3.  Winui3 will end up being the proper place for this fix to be made.

**Comment by groovykool** [comment]
But Winui3 has abandoned UWP.  It was my understanding that Winui2.x development was continuing?  I guess the correct answer is not really.

**Comment by groovykool** [comment]
Still a bug in ALL versions of Winui.

**Comment by groovykool** [comment]
Not fixed in any version of WinUi.......
